In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import classification_report, accuracy_score

In [13]:
df = pd.read_csv("D:/PROJECTS/Datasets/cleaned_titanic_data.csv")
print("✅ Dataset Loaded Successfully\n")
print(df.head())

✅ Dataset Loaded Successfully

   Survived  Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0         0       3    1  22.0      1      0   7.2500         2
1         1       1    0  38.0      1      0  71.2833         0
2         1       3    0  26.0      0      0   7.9250         2
3         1       1    0  35.0      1      0  53.1000         2
4         0       3    1  35.0      0      0   8.0500         2


In [15]:
X = df.drop("Survived", axis=1)  # Features
y = df["Survived"] 

In [17]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)


In [19]:
model = RandomForestClassifier(random_state=42)
model.fit(X_train, y_train)

RandomForestClassifier(random_state=42)

In [21]:
y_pred = model.predict(X_test)


In [23]:
print("\n✅ Classification Report:")
print(classification_report(y_test, y_pred))

print("✅ Accuracy Score:", accuracy_score(y_test, y_pred))


✅ Classification Report:
              precision    recall  f1-score   support

           0       0.83      0.88      0.85       105
           1       0.81      0.74      0.77        74

    accuracy                           0.82       179
   macro avg       0.82      0.81      0.81       179
weighted avg       0.82      0.82      0.82       179

✅ Accuracy Score: 0.8212290502793296


In this project, we used a cleaned Titanic dataset to build a machine learning model that predicts whether a passenger survived the disaster. After preprocessing the data by removing irrelevant features and encoding categorical values, we trained a Random Forest Classifier. The model was evaluated using accuracy and a classification report, achieving an accuracy of approximately 82%. This demonstrates the model’s ability to learn patterns from features like age, gender, class, fare, and embarkation point. The project showcases essential steps in a typical data science workflow: data cleaning, feature selection, model training, and evaluation.
